### PR TITLE
Add career ladder specification

### PR DESCRIPTION
**Pull request by damianavila**
This is based on the whole conversation in https://github.com/2i2c-org/meta/issues/402, the last few months of experience using some of these pieces (such as the matrix we used for the latest hiring processes), and conversations with team members.
I am purposely trying to reuse existing frameworks and build on top of it. This is why I am referencing some of those pieces via links (I do not think we should copy/paste the content but let me know if you disagree).
Any feedback is welcome!!

### CODE REVIEW

**Comment by sgibson91** [comment]
+1 (Not specifically approving so others can provide feedback too)

**Review by consideRatio — APPROVED** [review]
I think its a great call to adapt existing definitions of axes and levels etc instead of defining that ourselves - it seems very hard to do.

I figure future incremental improvements here could be to:
- define who evaluates step and level increments for the various track members
- define when step and level increments are evaluated

File: engineering/structure.md
Comment (damianavila) [reply engineering/structure.md:49]: Sure, I will add the additional radar plot.

**Comment by damianavila** [comment]
> I think its a great call to adapt existing definitions of axes and levels etc instead of defining that ourselves - it seems very hard to do.

Yep, that is the whole idea underneath this and follow-up PRs.

> I figure future incremental improvements here could be to:

Indeed!! After merging this PR the next step is the mapping of the current team in the proposed structure and then working on the processes for evaluation of levels/steps whereas we define short-term Q4 personal goals (before October) and medium-term 2024 goals (during Q4).

**Comment by damianavila** [comment]
@2i2c-org/engineering, I plan to merge this PR before EOW (tomorrow EOD) if I do not see any blockers. Please comment if you are inclined. Thanks! 

cc @GeorgianaElena and @yuvipanda

**Review by GeorgianaElena — APPROVED** [review]

**Review by yuvipanda — APPROVED** [review]
I am happy we are adapting an existing external ladder for the most part!

File: engineering/structure.md
Comment (damianavila) [reply engineering/structure.md]: Added in https://github.com/2i2c-org/team-compass/pull/765/commits/b92e07a9c44e36077742354462ae30acd3822e90

File: engineering/structure.md
Comment (damianavila) [inline engineering/structure.md:45]:
```suggestion
```{image} /images/em4.png
```

File: engineering/structure.md
Comment (damianavila) [inline engineering/structure.md:49]:
```suggestion
```{image} /images/em4-dark.png
```

**Pull request merged**
